python/lsst/resources/file.py

```diff
@@ -284,13 +284,39 @@ def transfer_from(
                     pass
 
             if transfer == "move":
-                with transaction.undoWith(
-                    f"move from {local_src}", shutil.move, newFullPath, local_src, copy_function=shutil.copy
-                ):
-                    shutil.move(local_src, newFullPath, copy_function=shutil.copy)
+                # If a rename works we try that since that is guaranteed to
+                # be atomic. If that fails we copy and rename. We do this
+                # in case other processes are trying to move to the same
+                # file and we want the "winner" to not be corrupted.
+                try:
+                    with transaction.undoWith(f"move from {local_src}", os.rename, newFullPath, local_src):
+                        os.rename(local_src, newFullPath)
+                except OSError:
```

Comment:
It's not easy to trigger this code path because the test environment does not have multiple disks where rename would fail.